### PR TITLE
Frontend feedback upvote + RTK changes

### PR DESCRIPTION
**Pull request by Elena-Erganyan**

### CODE REVIEW

File: client/src/components/Upvote/styled.ts [hunk 1]
```diff
@@ -3,13 +3,26 @@ import { setHeading4 } from '../Typo';
...
+export const StyledUpvote = styled(StyledLabel)<StyledUpvoteProps>`
   display: flex;
   flex-wrap: wrap;
   align-items: center;
   justify-content: center;
   gap: 0.8rem;
   min-width: 6.9rem;
+  background-color: ${({theme, isUpvoted}) => isUpvoted && theme.blue};
+
+  svg {
```
Comment (ALI-KHALED-13) [inline client/src/components/Upvote/styled.ts:19]: remove this, not doing anything

File: client/src/components/Upvote/styled.ts [hunk 2]
```diff
@@ -3,13 +3,26 @@ import { setHeading4 } from '../Typo';
...
+    fill: ${({theme, isUpvoted}) => isUpvoted && theme.white};
+  }
+
+  span {
```
Comment (ALI-KHALED-13) [inline client/src/components/Upvote/styled.ts:23]: you can pass isUpvoted to `StyledUpvoteNumber` instead and decide the color there based on the passed prop

File: client/src/components/Upvote/styled.ts [hunk 3]
```diff
@@ -3,13 +3,26 @@ import { setHeading4 } from '../Typo';
...
+    color: ${({theme, isUpvoted}) => isUpvoted && theme.white};
+  }
 
   &:active svg {
```
Comment (ALI-KHALED-13) [inline client/src/components/Upvote/styled.ts:27]: I would remove this too

File: client/src/App.tsx [hunk 4]
```diff
@@ -48,16 +48,16 @@ function App() {
...
               <Routes>
                   <Route path="/" element={<Navigate to={user? "/dashboard": "/landing"}/>}/>
                   <Route path="/dashboard" element={user? <Dashboard /> : <Navigate to="/"/>} />
                   <Route path="/landing" element={<Landing />} />
                   
                   <Route path="/products" element={<ProductsListing />} />
-                  <Route path="/products/:id" element={<ProductPage />} />
-                  <Route path="/products/:id/edit" element={<ProductForm />} />
+                  <Route path="/products/:productId" element={<ProductPage />} />
+                  <Route path="/products/:productId/edit" element={<ProductForm />} />
```
Comment (ALI-KHALED-13) [inline client/src/App.tsx:60]: please import Productform and Playground Lazily

File: client/src/pages/ProductForm/index.tsx [hunk 5]
```diff
@@ -5,11 +5,10 @@ import { useUsersSearchMutation } from "../../redux/api/userApi";
 import { StyledAddedContributer, StyledDeleteButton } from "./styled";
 import { X } from "@phosphor-icons/react";
 import Button from "../../components/Button";
-import { useAddProductMutation, useDeleteProductMutation, useEditProductMutation } from "../../redux/api/productsApi";
+import { useAddProductMutation, useDeleteProductMutation, useEditProductMutation, useGetProductByIdQuery } from "../../redux/api/productsApi";
```
Comment (ALI-KHALED-13) [inline client/src/pages/ProductForm/index.tsx:8]: readability note: edit this into multiple lines

File: client/src/redux/api/feedbackApi.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        body: data,
+      }),
+      async onQueryStarted({ id, data }, { dispatch, queryFulfilled }) {
+        const patchResult = dispatch(
+          feedbackApi.util.updateQueryData('getFeedbackList', {key: 'targetProduct', value: data.targetProduct as string}, draft => {
+  
+            const currentFeedback = draft.find((item) => item._id === id);
+
+            if (currentFeedback) {
+              Object.assign(currentFeedback, data);
```
Comment (ALI-KHALED-13) [inline client/src/redux/api/feedbackApi.ts:42]: well done :)

File: client/src/redux/api/feedbackApi.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+const feedbackApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getFeedbackList: builder.query<IFeedback[], {key: string, value: string}>({
+      query: ({key, value}) => `feedback?${key}=${value}`,
+      keepUnusedDataFor: 300,
+      providesTags: (result) =>
+        result
+          ? [
+              ...result.map(({ _id }) => ({ type: 'Feedback' as const, id: _id })),
```
Comment (ALI-KHALED-13) [inline client/src/redux/api/feedbackApi.ts:11]: what's the point of having a key/tag for each feedback?

**Review by ALI-KHALED-13 — CHANGES_REQUESTED** [review]
well done overall, the main work of upvoting and its logic is perfect.
left some comments about random stuff here and there

File: client/src/components/RoadmapInfo/index.tsx [hunk 8]
```diff
@@ -39,10 +39,10 @@ const RoadmapInfo = () => {
       </StyledRoadmapHeader>
 
       <StyledStatusList>
-        {isLoading
+        {isLoading && isFeedbackLoading
           ? <div>Loading...</div>
```
[the remark this answers is not in the document]
Comment (Elena-Erganyan) [reply client/src/components/RoadmapInfo/index.tsx:43]: We will have a skeleton loader here

File: client/src/components/Upvote/styled.ts (hunk 2 above)
Comment (Elena-Erganyan) [reply client/src/components/Upvote/styled.ts:23]: Changed

File: client/src/components/Upvote/styled.ts (hunk 3 above)
Comment (Elena-Erganyan) [reply client/src/components/Upvote/styled.ts:27]: Not removing it, when you click the upvote button the arrow should be white

File: client/src/redux/api/feedbackApi.ts (hunk 7 above)
Comment (Elena-Erganyan) [reply client/src/redux/api/feedbackApi.ts:11]: From what I get it means that if one feedback is updated or deleted (i.e. have `invalidateKeys: (result) => [type: 'Feedback', id: result._id]` then this feedback list will be re-fetched.
We have optimistic updates, so in our case it's needed when we delete one feedback item only.

**Pull request merged**